### PR TITLE
feat!: Update job to include removing macra measures

### PR DESCRIPTION
**Pull request by tmy1313**
OCD-3221

### CODE REVIEW

File: chpl/chpl-resources/src/main/resources-production/jobs.xml [hunk 1]
```diff
@@ -12,11 +12,11 @@
     </pre-processing-commands>
     <schedule>
         <!-- Delete from here to "DELETE END MARKER" after PROD push -->
-        <!--  Updates all g1g2 "ACI" MACRA measures to "PI" -->
+        <!--  Updates all MACRA measures to based on new requirements -->
         <job>
             <name>updateMacraMeasuresJob</name>
             <group>systemJobs</group>
-            <description>Updates all g1g2 ACI MACRA measures to PI</description>
+            <description>Updates all MACRA measures to based on new requirements</description>
```
Comment (andlar) [inline chpl/chpl-resources/src/main/resources-production/jobs.xml]: If the previous job goes to PROD before this one does we'll need to make sure that updating a job description in the jobs.xml for PROD works correctly

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/UpdateMacraMeasuresJob.java [hunk 2]
```diff
@@ -73,15 +92,27 @@ protected void doInTransactionWithoutResult(TransactionStatus status) {
...
+                    .forEach(measure -> {
+                        measure.setValue(measure.getValue().replaceAll(oldSubstring, newSubstring));
+                        entityManager.merge(measure);
+                    });
+        }
+
+        private void updateStage2AsRemoved() {
+            getAllMeasuresWhereValueIsNotNull().stream()
+                    .filter(measure -> (measure.getCertificationCriterion().getNumber().equals(G_8)
+                            || measure.getCertificationCriterion().getNumber().equals(G_9))
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/UpdateMacraMeasuresJob.java]: I know there is a Cures version of g9. I am not sure the order the jobs will be/need to be run but I just want to make sure that this code works as intended whether there is one g9 or two.

File: chpl/chpl-resources/src/main/resources-dev/jobs.xml [hunk 3]
```diff
@@ -902,11 +902,11 @@
                 </entry>
             </job-data-map>
         </job>
-        <!--  Updates all g1g2 "ACI" MACRA measures to "PI" -->
+        <!-- Updates all MACRA measures to based on new requirements -->
```
Comment (drbgfc) [inline chpl/chpl-resources/src/main/resources-dev/jobs.xml]: Suggest removing 'to', grammar. Remove in the description tag too, and everywhere else it was copied.

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/UpdateMacraMeasuresJob.java [hunk 4]
```diff
@@ -31,6 +34,102 @@
...
+    private static final String E_1 = "170.315 (e)(1)";
+    private static final String E_2 = "170.315 (e)(2)";
+    private static final String G_8 = "170.315 (g)(8)";
+    private static final String G_9 = "170.315 (g)(9)";
+
+    private static Map<String, List<String>> measuresToRemove;
+    static {
+        measuresToRemove = new HashMap<String, List<String>>();
+        measuresToRemove.put(A_1,
+                Arrays.asList("EP Stage 2",
```
Comment (drbgfc) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/UpdateMacraMeasuresJob.java]: There are plenty of repeats of Strings within these Arrays which could be externalized as they leave room for error as is

**Pull request merged**